Game/MyProject/Game/GameStates/RaceState.cpp

```diff
@@ -30,10 +37,212 @@ void RaceState::onExit() {
+}
+
+bool RaceState::isGameOverTriggered()
+{
+    return IvGame::mGame->mEventHandler->IsKeyPressed('q');
 }
 
-bool RaceState::isQuitTriggered() {
-    return IvGame::mGame->mEventHandler->IsKeyDown('q');
+unsigned int RaceState::noDigits(uint32_t number)
+{
+    if(number<10)
+        return 1;
+    else
+        return 1+noDigits(number/10);
+}
+
+void RaceState::addNewDigit()
+{
+        //create a new quad
+        std::vector<std::string> uniforms;
+        uniforms.push_back("mTexture");
+        uniforms.push_back("row");
+        uniforms.push_back("column");
+    
+        std::string name="count"+std::to_string(noDigits(score));
+    
+        const char* quadName=name.c_str();
+    
+        std::string preName="count"+std::to_string(noDigits(score-1));
+    
+        const char* previousQuadName=preName.c_str();
+        
+        std::shared_ptr<SceneNode> countScoreQuad=HelperManager::BuildTexturedQuad(quadName,HelperManager::CreateMeshInstance(meshManager.GetMesh("quad"),uniforms,"../../Game/BasicMenu/Shaders/AtlasSpriteShader"),"font.tga");
+        
+        auto previousPos=raceMenu->getScene()->getRoot()->findFirstNodeWithName(previousQuadName)->getLocalPosition();
+        countScoreQuad->setLocalTransform(IvVector3{previousPos.x-1,previousPos.y,previousPos.z}, IvVector3{0,4.72,1}, IvVector3{2,2,2});
+        
+        raceMenu->getScene()->getRoot()->findFirstNodeWithName("countRoot")->addChild(countScoreQuad);
+    
+        auto val=raceMenu->getRowCol(score%10 +'0');
+    
+        name="count"+std::to_string(noDigits(score%10));
+    
+        quadName=name.c_str();
+    
+        raceMenu->getScene()->getRoot()->findFirstNodeWithName(quadName)->getRenderable()->setUniformValue(1, val.first);
+        raceMenu->getScene()->getRoot()->findFirstNodeWithName(quadName)->getRenderable()->setUniformValue(2, val.second);
+    
+        if(score>=10)
+        {
+            val=raceMenu->getRowCol(firstDigit(score)+'0');
+    
+            name="count"+std::to_string(noDigits(score));
+    
+            quadName=name.c_str();
+    
+            raceMenu->getScene()->getRoot()->findFirstNodeWithName(quadName)->getRenderable()->setUniformValue(1, val.first);
+            raceMenu->getScene()->getRoot()->findFirstNodeWithName(quadName)->getRenderable()->setUniformValue(2, val.second);
+        }
+    
+}
+
+
+void RaceState::checkNeedChangeDigit()
+{
+    auto aux =  score-1;
+    auto noDigits = 1;
+    auto continue_ = false;
+    
+    while(aux!=0)
+    {
+        continue_=false;
+        
+        if(aux%10==9 )
+        {
+            continue_=true;
+            
+            std::string name="count"+std::to_string(noDigits+1);
+            
+            const char* quadName=name.c_str();
+            
+            char ch=(aux/10%10+1)%10 +'0';
+        
+            auto val=raceMenu->getRowCol(ch );
+            if(raceMenu->getScene()->getRoot()->findFirstNodeWithName(quadName)!=nullptr)
```

Comment:
`findFirstNodeWithName` will get called three times in this context, redundant operation.
```suggestion
           auto some_node = raceMenu->getScene()->getRoot()->findFirstNodeWithName(quadName);
           if(some_node != nullptr)
           {
                    some_node->setUniform(...);
           }
```